### ISSUE TITLE
Shouldn't BatchNorm be applied after the activations? (DCGAN)

### ISSUE THREAD

**Issue by aybberk** [pytorch/examples#383]
The original paper about BatchNorm argues it should be applied after non-linearities to make it actually reduce covariate shift. I couldn't find anything in the DCGAN paper about BatchNorm being at before or after nonlinearities but here (and at the LuaTorch implementation) it is being applied before the activations. Was that a choice after empirical evidences when you were searching along the topological options or does it really matter at all?

**Comment by soumith** [comment]
empirically it didn't matter, we just went with "before" probably without any particular reason.
